### PR TITLE
Bug 1120491 - Include a test that all of the addon-sdk files use mpl 2 headers

### PR DESCRIPTION
**Pull request by erikvold**

### CODE REVIEW

File: lib/sdk/system/child_process/subprocess.js [hunk 1]
```diff
@@ -1,40 +1,6 @@
...
- * under the terms of either the GPL or the LGPL, and not to allow others to
- * use your version of this file under the terms of the MPL, indicate your
- * decision by deleting the provisions above and replace them with the notice
- * and other provisions required by the GPL or the LGPL. If you do not delete
- * the provisions above, a recipient may use your version of this file under
- * the terms of any one of the MPL, the GPL or the LGPL.
- * ***** END LICENSE BLOCK ***** */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (Gozala) [inline lib/sdk/system/child_process/subprocess.js]: We can't change license of code we've imported that has a diff license.

File: lib/sdk/system/child_process/subprocess_worker_unix.js [hunk 2]
```diff
@@ -1,38 +1,6 @@
...
- * under the terms of either the GPL or the LGPL, and not to allow others to
- * use your version of this file under the terms of the MPL, indicate your
- * decision by deleting the provisions above and replace them with the notice
- * and other provisions required by the GPL or the LGPL. If you do not delete
- * the provisions above, a recipient may use your version of this file under
- * the terms of any one of the MPL, the GPL or the LGPL.
- * ***** END LICENSE BLOCK ***** */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (Gozala) [inline lib/sdk/system/child_process/subprocess_worker_unix.js]: We can't change license of code that isn't ours.

File: lib/sdk/system/child_process/subprocess_worker_win.js [hunk 3]
```diff
@@ -1,38 +1,6 @@
...
- * under the terms of either the GPL or the LGPL, and not to allow others to
- * use your version of this file under the terms of the MPL, indicate your
- * decision by deleting the provisions above and replace them with the notice
- * and other provisions required by the GPL or the LGPL. If you do not delete
- * the provisions above, a recipient may use your version of this file under
- * the terms of any one of the MPL, the GPL or the LGPL.
- * ***** END LICENSE BLOCK ***** */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (Gozala) [inline lib/sdk/system/child_process/subprocess_worker_win.js]: We can't change license of code that is not ours.

File: lib/sdk/system/child_process/subprocess.js (hunk 1 above)
Comment (Mossop) [inline lib/sdk/system/child_process/subprocess.js]: Gozala and Gerv's comments seem to stand here. Where did this code come from and is there a good reason to change it to MPL2?

File: test/test-mpl2-license-header.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+const options = require('@loader/options');
+const { id } = require("sdk/self");
+const { getAddonByID } = require("sdk/addon/manager");
+const { mapcat, map, filter, fromEnumerator } = require("sdk/util/sequence");
+const { readURISync } = require('sdk/net/url');
+
+const ios = Cc['@mozilla.org/network/io-service;1'].
+              getService(Ci.nsIIOService);
+
+ const MIT_LICENSE_HEADER = [];
```
Comment (Mossop) [inline test/test-mpl2-license-header.js]: Nit: additional space

File: test/test-mpl2-license-header.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+const uri = (path="") => baseURI + path;
+
+const toFile = x => x.QueryInterface(Ci.nsIFile);
+const isTestFile = ({ path, leafName }) => {
+  return !ignoreFile(path) && /\.jsm?$/.test(leafName)
+};
+const getFileURI = x => ios.newFileURI(x).spec;
+
+const getDirectoryEntries = file => map(toFile, fromEnumerator(_ => file.directoryEntries));
```
Comment (Mossop) [inline test/test-mpl2-license-header.js]: Seems like it might be useful to have fromEnumerator take an iid to QI everything to

**Comment by Mossop** [comment]
This test will probably fail in mochitest-jetpack so it should be marked disabled in jetpack-package.ini.

File: lib/sdk/system/child_process/subprocess.js (hunk 1 above)
Comment (erikvold) [inline lib/sdk/system/child_process/subprocess.js]: ah good catch, I'll switch it back

File: test/test-mpl2-license-header.js (hunk 5 above)
Comment (erikvold) [inline test/test-mpl2-license-header.js]: Let's make that a different bug then, I'm pretty sure I copied this function from the deprecated/unit-test code

Comment (Mossop) [inline test/test-mpl2-license-header.js]: Sure

**Pull request merged**